packages/frontend-2/pages/workspaces/[id]/index.vue

```diff
@@ -4,12 +4,37 @@
   </div>
 </template>
 <script setup lang="ts">
+import { useQuery } from '@vue/apollo-composable'
 import { useRoute } from 'vue-router'
+import { useWorkspacesMixpanel } from '~/lib/workspaces/composables/mixpanel'
+import { workspaceMixpanelUpdateGroupQuery } from '~/lib/workspaces/graphql/queries'
 
 const route = useRoute()
 const workspaceId = computed(() => route.params.id as string)
+const { workspaceMixpanelUpdateGroup } = useWorkspacesMixpanel()
+const pageFetchPolicy = usePageQueryStandardFetchPolicy()
 
 definePageMeta({
   middleware: ['requires-workspaces-enabled', 'require-valid-workspace']
 })
+
+const { result } = useQuery(
+  workspaceMixpanelUpdateGroupQuery,
+  () => ({
+    workspaceId: workspaceId.value
+  }),
+  () => ({
+    fetchPolicy: pageFetchPolicy.value
+  })
+)
+
+watch(
```

Comment:
Instead of (deep) watching, just run the updater once onResult. If there are any changes to the workspace (e.g. versionCount), you're not watching for them anyway (you only run updater if workspaceId changes) and you will get those changes the next time the page is reloaded.